lib/elixir/src/elixir_clauses.erl

```diff
@@ -401,6 +405,8 @@ format_error({try_with_only_else_clause, Origin}) ->
 
 format_error(unmatchable_else_in_with) ->
   "\"else\" clauses will never match because all patterns in \"with\" will always match";
+format_error(non_match_argument_in_with) ->
+  "\"with\" arguments should be matches, not statements";
```

Comment:
```suggestion
  "\"with\" requires match expressions to be given either as \"left <- right\" or \"left = right\", "
  "please assign your expression to underscore if you want to simply execute some code and discard its value"
```

WDYT?

Reply:
Looks great, thanks!

I just realized I forgot to trap the warning output of my test. I'll use this new warning text and push a new commit.